### ISSUE TITLE
Allow implementing custom brush types

### ISSUE THREAD

**Issue by mysticfall** [SixLabors/ImageSharp#970]
For now, `IBrush` is supposed to return an instance of `BrushApplicator`, but both its constructor and indexer are defined with an `internal` modifier, which makes it practically impossible to create a custom brush implementation.

It can prevent users from adopting ImageSharp in case they need a type of brush that the library doesn't support, as described in #969, for example.

**Comment by JimBobSquarePants** [comment]
Hi @mysticfall 

We've currently been keeping those APIs internal as we expect some churn as we separate out and optimize the drawing portion of the library, however we would be very happy work with you if you would like to implement a PR against the main library and add the PathGradientBrush described in #969

**Comment by antonfirsov** [comment]
Yeah those API-s are very unstable, many changes are expected, and we try to avoid breaking changes.

**Comment by mysticfall** [comment]
@antonfirsov Thanks for the pointer. I'm still undecided what to do with the situation, as serializing textures was initially only a small part of the API which I've been writing.

Not sure if I could justify spending much time in creating such an implementation that can be accepted to be merged (i.e. with studying conventions, writing tests, and etc.) and postponing the main project until it gets released.

I'm not complaining, and I'd love to contribute if I could. It's just that I can only spend only small amount of time for my own project for which texture serialization is but a small part.

I will take a serious look at the code to see if I can make a PR quickly though.

**Comment by antonfirsov** [comment]
@mysticfall yeah we are quite strict with both code style and test, but we can support you with those, if you make up your mind to contribute. I'm even happy to push code changes to your PR branch, if it's not too much & I have the time.

**Comment by mysticfall** [comment]
@antonfirsov I opened a pull request for #969, and I think I need some help or an advice as to what to do with test fixtures which belongs to a different repository(submodule).

I'd appreciate if you could help me resolving this problem.

**Comment by JimBobSquarePants** [comment]
@mysticfall Will have a good look for you as soon as possible. Thanks so much for your work so far!

**Comment by antonfirsov** [comment]
@mysticfall thanks a lot, for investing your time! Looks like @JimBobSquarePants has it under control, but ping me if I can help.
